### ISSUE TITLE
Legends on pie charts

### ISSUE THREAD

**Issue by nutjob4life** [EDRN/P5#168]
From @dcrichto:

> Another thing.  If I look at protocols, the pie charts don’t have a legend, right?   I can hover over them but seems like we should be able to see what they are without doing that.

**Comment by hoodriverheather** [comment]
@nutjob4life this seems to be the case for Data and Biomarkers too.

**Comment by nutjob4life** [comment]
@hoodriverheather correct, but the pie chart on the Biomarkers page does have a legend.

On science data and protocols, there is no legend.

**Comment by nutjob4life** [comment]
@hoodriverheather this is fixed:

- Science data: https://edrn-dev.jpl.nasa.gov/portal/renaissance/data-and-resources/data/
- Protocols: https://edrn-dev.jpl.nasa.gov/portal/renaissance/data-and-resources/protocols/

I had to stack the pie charts vertically because the legends take up so much space.

**Comment by hoodriverheather** [comment]
@nutjob4life If they have to be stacked vertically, do you think we should keep all the charts? I have to scroll down to view the list, search titles, etc.

**Comment by nutjob4life** [comment]
@hoodriverheather I don't know what the right answer is. I can't shrink the charts without violating Section 508 minimum font sizes. Personally, I liked them _without_ the legends  because ① they fit, and ② they encourage interaction and exploration.

We can hide the charts under a kind of accordion, but I think accordions have to be vertical. I'll have to check if we can do horizontal: show just one chart at a time, and you can swap left/right to other charts.

**Comment by hoodriverheather** [comment]
@nutjob4life Dang! Maybe good to let Dan know the options and he can decide :)

**Comment by nutjob4life** [comment]
I wish @dcrichto would join the EDRN organization so we could just `@` him!

**Comment by nutjob4life** [comment]
Oof. This was really hard.

Plotly has built-in legend support as you've seen, but it considers legends to be part of the chart (since they're interactive: you can toggle items on the legend). However, although you can anchor the legend to certain directions, because they're part of the chart, each pie ends up being a different size. You can specify a size for the chart, not for the pie, and because each pie's legends have different items, they end up changing the overall display.

I had to go back to disabling the legends, specifying a color palette, and then manually generate each legend beneath each pie chart (and due to the way Plotly and Django don't like to interact, I had to generate the legends in the view method for the content type instead of in the content type's template and give it to Plotly to render).

**Comment by hoodriverheather** [comment]
@nutjob4life It seems like totals in the top should only include the  "Consortium" = "EDRN".
I agree with Dan and don't understand what count is when you hover over.
I also wonder if the very bright purple/pink color could be something different? It's very bright :)

**Comment by hoodriverheather** [comment]
Pie charts and legends look great after removing one of the charts, etc. based on Dan's suggestions.
